### PR TITLE
[KOGITO-5054] Review pom configuration

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/KOGITO-5054

- Created `kogito-build` module to contain `kogito-build-parent` and `kogito-ide-config`
- Removed many unused dependencies from dependency management
- Removed some platform specific dependencies to reduce kogito core "footprint": we had multiple dependencies declared in kogito-built-parent that are platform specific (i.e. kafka version or restAssured) and that are already overridden by importing quarkus-bom/spring-dependencies
- Removed dependencies used only in `kogito-apps` ~~(to be moved there as part of this PR)~~ Done

Related PRs
- https://github.com/kiegroup/kogito-apps/pull/789
- https://github.com/kiegroup/kogito-runtimes/pull/1210
- https://github.com/kiegroup/kogito-examples/pull/668

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Review by evacchi — APPROVED** [review]
beautiful cleanup. Let's hope CI is green

**Comment by danielezonca** [comment]
jenkins retest this please

File: kogito-build/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  <artifactId>kogito-build</artifactId>
+  <packaging>pom</packaging>
+
+  <name>Kogito :: Build Modules</name>
+
+  <profiles>
+    <profile>
+      <id>default</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```
Comment (cristianonicolai) [inline kogito-build/pom.xml:23]: @danielezonca dont we have to avoid this active by default because of IntelliJ?

Comment (danielezonca) [reply kogito-build/pom.xml]: My bad, I originally created this PR before that change so fixed now 👍

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by evacchi** [comment]
@danielezonca conflict

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
Network problem

jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Review by evacchi — APPROVED** [review]

File: integration-tests/integration-tests-springboot/pom.xml [hunk 2]
```diff
@@ -24,104 +19,6 @@
...
-    <!-- Force resolving dependencies in local Maven repo, needed for it tests -->
-    <dependency>
-       <groupId>io.rest-assured</groupId>
-       <artifactId>json-schema-validator</artifactId>
-       <!-- Manually override to 3.3.0 because Spring boot uses rest-assured-common internally -->
-       <version>${version.io.rest-assured.springboot}</version>
-       <scope>test</scope>
-    </dependency>
-
-    <dependency>
```
Comment (jstastny-cz) [inline integration-tests/integration-tests-springboot/pom.xml:43]: Is it safe to remove these? I've explicitly added these test deps for maven reactor to be in order: https://github.com/kiegroup/kogito-runtimes/pull/1188/files#diff-48f36d228f186a4252f88c2ebd6cdc3500608ea5117846970543a58b2ced7e9b

**Review by jstastny-cz — CHANGES_REQUESTED** [review]
I am mostly concerned about invoker-based tests - there are removals of explicit test dependencies that make sure reactor is in good shape (to assure deps of src/it projects are in reactor and not relying on previously released snapshots in some repository). I think here we're about to reach the situation of not being able to run full build in env detached from our nexus (like quarkus CI for example).

File: integration-tests/integration-tests-springboot/pom.xml (hunk 2 above)
Comment (danielezonca) [reply integration-tests/integration-tests-springboot/pom.xml:43]: You are right, restored 👍

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
@jstastny-cz 
Can you please review again?

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-norest-it/pom.xml [hunk 3]
```diff
@@ -73,23 +71,19 @@
             <artifactId>junit-jupiter-engine</artifactId>
             <scope>test</scope>
         </dependency>
-        
+
         <dependency>
             <groupId>io.rest-assured</groupId>
             <artifactId>json-schema-validator</artifactId>
-            <!-- Manually override to 3.3.0 because Spring boot uses rest-assured-common internally -->
```
Comment (cristianonicolai) [inline integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-norest-it/pom.xml:80]: @danielezonca if we remove this, what version would be pick? or is just that versions are aligned and this is not needed any more.

**Review by cristianonicolai — APPROVED** [review]

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-norest-it/pom.xml (hunk 3 above)
Comment (danielezonca) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-norest-it/pom.xml:80]: RestAssured is both declared in `quarkus-bom` and in `spring-boot-dependencies` so the compatible version is automatically provided with the platform: "kogito-core" doesn't really need RestAssured, it is only required to perform e2e test on a specific platform so in this way everything works automatically (same apply to kafka dependency)

Comment (cristianonicolai) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-norest-it/pom.xml:80]: makes sense, thanks

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by manstis** [comment]
Did I miss the email communicating the move of the IDE configuration files?

**Comment by danielezonca** [comment]
> Did I miss the email communicating the move of the IDE configuration files?

You are right, it took ages to get a green build and when I merged it I forgot to update the team, email sent now :)

**Pull request merged**